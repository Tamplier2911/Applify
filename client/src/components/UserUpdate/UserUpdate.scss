/*

.user-update {
  display: grid;

  &__info {
    display: grid;
    grid-template-columns: 1fr min-content;
    grid-column-gap: 2rem;
    padding: 1rem;
    font-family: var(--font-title);

    @media only screen and (max-width: 650px) {
      grid-template-columns: 1fr;
      grid-row-gap: 2rem;
      justify-items: center;
    }
  }
  &__credentials {
    display: grid;

    &--name {
      display: grid;
      grid-template-columns: 3fr 9fr;

      @media only screen and (max-width: 650px) {
        grid-template-columns: repeat(2, 1fr);
      }

      @media only screen and (max-width: 425px) {
        grid-template-columns: 1fr;
        text-align: center;
      }

      & div:nth-child(1) {
        display: grid;
        font-weight: bold;
      }

      & div:nth-child(2) {
        display: grid;
        width: 100%;
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
      }
    }

    &--email {
      display: grid;
      grid-template-columns: 3fr 9fr;

      @media only screen and (max-width: 650px) {
        grid-template-columns: repeat(2, 1fr);
      }

      @media only screen and (max-width: 425px) {
        grid-template-columns: 1fr;
        text-align: center;
      }

      & div:nth-child(1) {
        display: grid;
        font-weight: bold;
      }

      & div:nth-child(2) {
        display: grid;
        width: 100%;
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
      }
    }

    &--role {
      display: grid;
      grid-template-columns: 3fr 9fr;

      @media only screen and (max-width: 650px) {
        grid-template-columns: repeat(2, 1fr);
      }

      @media only screen and (max-width: 425px) {
        grid-template-columns: 1fr;
        text-align: center;
      }

      & div:nth-child(1) {
        display: grid;
        font-weight: bold;
      }
    }

    &--id {
      display: grid;
      grid-template-columns: 3fr 9fr;

      @media only screen and (max-width: 650px) {
        grid-template-columns: repeat(2, 1fr);
      }

      @media only screen and (max-width: 425px) {
        grid-template-columns: 1fr;
        text-align: center;
      }

      & div:nth-child(1) {
        display: grid;
        font-weight: bold;
      }
    }

    &--changed {
      display: grid;
      grid-template-columns: 3fr 9fr;

      @media only screen and (max-width: 650px) {
        grid-template-columns: repeat(2, 1fr);
      }

      @media only screen and (max-width: 425px) {
        grid-template-columns: 1fr;
        text-align: center;
      }

      & div:nth-child(1) {
        display: grid;
        font-weight: bold;
      }
    }
  }
  &__photo {
    display: grid;
    align-content: center;

    &--wrapper {
      width: 10rem;
      height: 10rem;
      border-radius: 50%;
      overflow: hidden;
      box-shadow: 0rem 0rem 0.8rem var(--cl-primary);
    }
    &--img {
      width: 100%;
      height: 100%;
      display: block;
      object-fit: cover;
    }
  }
}

*/
