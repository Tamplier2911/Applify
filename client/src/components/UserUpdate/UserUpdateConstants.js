const userUpdateData = {
  eng: {
    userUpdateName: "Name:",
    userUpdateEmail: "Email:",
    userUpdateRole: "Role:",
    userUpdateId: "ID:",
    userUpdatePassword: "Password Changed:",
    userUpdatePasswordChanged: "Never changed",
    userUpdateNotFound: "Object with that id not found."
  },
  rus: {
    userUpdateName: "Имя:",
    userUpdateEmail: "Эл. Адрес:",
    userUpdateRole: "Роль:",
    userUpdateId: "ИД:",
    userUpdatePassword: "Пароль изменен:",
    userUpdatePasswordChanged: "Никогда не менялся",
    userUpdateNotFound: "Объект с таким идентификатором не найден."
  },
  ukr: {
    userUpdateName: "Ім'я:",
    userUpdateEmail: "Ел. Адреса:",
    userUpdateRole: "Роль::",
    userUpdateId: "ІД:",
    userUpdatePassword: "Пароль змінений:",
    userUpdatePasswordChanged: "Ніколи не змінювався",
    userUpdateNotFound: "Об'єкт з таким ідентифікатором не найден."
  }
};

export default userUpdateData;
