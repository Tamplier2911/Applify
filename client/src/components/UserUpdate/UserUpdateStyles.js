import styled, { css } from "styled-components";

const sharedFieldStyles = css`
  display: grid;
  grid-template-columns: 3fr 9fr;

  @media only screen and (max-width: 650px) {
    grid-template-columns: repeat(2, 1fr);
  }

  @media only screen and (max-width: 425px) {
    grid-template-columns: 1fr;
    text-align: center;
  }
`;

const sharedWrappingOptions = css`
  display: grid;
  width: 100%;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;

export const UserUpdateContainer = styled.div`
  display: grid;
  margin-bottom: 5rem;
`;

export const UserUpdateInfo = styled.div`
  display: grid;
  grid-template-columns: 1fr min-content;
  grid-column-gap: 2rem;
  padding: 1rem;
  font-family: var(--font-title);

  @media only screen and (max-width: 650px) {
    grid-template-columns: 1fr;
    grid-row-gap: 2rem;
    justify-items: center;
  }
`;

export const UserUpdateCred = styled.div`
  display: grid;
`;

export const UserUpdateName = styled.div`
  ${sharedFieldStyles}
`;

export const UserUpdateEmail = styled.div`
  ${sharedFieldStyles}
`;

export const UserUpdateRole = styled.div`
  ${sharedFieldStyles}
`;

export const UserUpdateId = styled.div`
  ${sharedFieldStyles}
`;

export const UserUpdateChanged = styled.div`
  ${sharedFieldStyles}
`;

export const UserUpdatePhoto = styled.div`
  display: grid;
  align-content: center;
`;

export const UserUpdatePhotoWrap = styled.div`
  width: 10rem;
  height: 10rem;
  border-radius: 50%;
  overflow: hidden;
  box-shadow: 0rem 0rem 0.8rem var(--cl-primary);
`;

export const UserUpdatePhotoImg = styled.img`
  width: 100%;
  height: 100%;
  display: block;
  object-fit: cover;
`;

export const UserFieldKey = styled.div`
  display: grid;
  font-weight: bold;
`;

export const UserFieldValue = styled.div`
  ${sharedWrappingOptions}
`;

export const UserUpdateNotFound = styled.div`
  font-family: var(--font-title);
  line-height: 1;
  font-size: 4rem;
  font-weight: 300;
`;
