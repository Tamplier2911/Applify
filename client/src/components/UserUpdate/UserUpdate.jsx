import "./UserUpdate.scss";
import React from "react";
import { withRouter } from "react-router-dom";

// redux
import { connect } from "react-redux";
import { selectAllUsersAsObject } from "../../redux/users/users.selectors";

// components
import GetBack from "../GetBack/GetBack";
import CreateUpdateUser from "../CreateUpdateUser/CreateUpdateUser";

// data formaters
import getImageRelativePath from "../../utils/PathTransformations/getImageRelativePath";
import transformDateToLocaleString from "../../utils/DataTransformations/transformDateToLocaleString";

// JS Rendering CSS
import {} from "./UserUpdateStyles";

// component constants

const UserUpdate = ({ userObject }) => {
  const { _id, photo, about, role, name, email, passwordChangedAt } = userObject
    ? userObject
    : {};

  const image = getImageRelativePath(photo ? photo : "");
  const date = transformDateToLocaleString(
    passwordChangedAt ? passwordChangedAt : ""
  );

  return userObject ? (
    <div className="user-update">
      <div>Name: {name}</div>
      <div>Email: {email}</div>
      <div>Role: {role}</div>
      <div>About: {about}</div>
      <div>ID: {_id}</div>
      <div>Password Changed: {date ? date : "Never changed"}</div>
      <div
        style={{
          width: "10rem",
          height: "10rem",
          borderRadius: "50%",
          overflow: "hidden",
          boxShadow: `0rem 0rem .8rem var(--cl-primary)`
        }}
      >
        <img
          src={image}
          alt="happy user"
          style={{
            width: "100%",
            height: "100%",
            display: "block",
            objectFit: "cover"
          }}
        ></img>
      </div>
      <CreateUpdateUser method="PATCH" />
      <GetBack path={`/profile/users`} />
    </div>
  ) : (
    <div>Object with that id not found.</div>
  );
};

const mapStateToProps = (state, ownProps) => ({
  userObject: selectAllUsersAsObject(ownProps.match.params.id)(state)
});

export default withRouter(connect(mapStateToProps)(UserUpdate));
