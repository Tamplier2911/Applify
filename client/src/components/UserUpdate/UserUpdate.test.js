import React from "react";
import toJson from "enzyme-to-json";

// component
import UserUpdate from "./UserUpdate";

// utils
import {
  setShallowWRC,
  findByTestAttr,
  storeFactory,
} from "../../tests/testUtils";

describe("<UserUpdate />", () => {
  const initialState = {
    lang: {
      languagePanelHidden: true,
      currentLanguage: "eng",
    },
    users: {
      allUsers: [
        {
          about: "Wonderful!",
          email: "sussy@gmail.com",
          likedBlogposts: [],
          name: "Suss",
          photo: "",
          role: "user",
          _id: "testObjSdk",
        },
      ],
      isLoading: false,
      errorMessage: null,
    },
  };
  const store = storeFactory(initialState);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowWRC(
      <UserUpdate store={store} testObjId="testObjSdk" />
    );
    console.log(wrapper.debug());
  });

  it("renders without an error", () => {
    // const userUpdate = findByTestAttr(wrapper, "user-update");
    // expect(userUpdate).toHaveLength(1);
  });

  it("renders children - user update info", () => {
    // const userUpdateInfo = findByTestAttr(wrapper, "user-update-info");
    // expect(userUpdateInfo).toHaveLength(0);
  });

  it("renders children - user update create/update", () => {
    // const userUpdateCu = findByTestAttr(wrapper, "user-update-cu");
    // expect(userUpdateCu).toHaveLength(0);
  });

  it("renders fallback when user object is not found", () => {
    // const userUpdate = findByTestAttr(wrapper, "user-update");
    // expect(userUpdate).toHaveLength(0);
  });

  it("renders without an error", () => {
    // const userUpdate = findByTestAttr(wrapper, "user-update");
    //     expect(toJson(userUpdate)).toMatchSnapshot();
  });
});
