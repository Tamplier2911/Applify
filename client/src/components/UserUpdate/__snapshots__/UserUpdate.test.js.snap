// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserUpdate /> renders without an error 1`] = `
<styled.div
  data-test="user-update"
>
  <styled.div
    data-test="user-update-info"
  >
    <styled.div>
      <styled.div>
        <styled.div>
          Name:
        </styled.div>
        <styled.div>
          Suss
        </styled.div>
      </styled.div>
      <styled.div>
        <styled.div>
          Email:
        </styled.div>
        <styled.div>
          sussy@gmail.com
        </styled.div>
      </styled.div>
      <styled.div>
        <styled.div>
          Role:
        </styled.div>
        <styled.div>
          user
        </styled.div>
      </styled.div>
      <styled.div>
        <styled.div>
          ID:
        </styled.div>
        <styled.div>
          testObjSdk
        </styled.div>
      </styled.div>
      <styled.div>
        <styled.div>
          Password Changed:
        </styled.div>
        <styled.div>
          Never changed
        </styled.div>
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.img
          alt="happy user"
          src="http://localhost/api/uploads/images/users/default.jpg"
        />
      </styled.div>
    </styled.div>
  </styled.div>
  <Connect(CreateUpdateUser)
    data-test="user-update-cu"
    method="PATCH"
    updateData={
      Object {
        "_id": "testObjSdk",
        "about": "Wonderful!",
        "email": "sussy@gmail.com",
        "name": "Suss",
        "role": "user",
      }
    }
  />
  <withRouter(GetBack)
    path="/profile/users"
  />
</styled.div>
`;
